Resources/Computing/ARCComputingElement.py

```diff
@@ -291,19 +294,49 @@ def getCEStatus( self ):
     if not result['OK']:
       gLogger.error( 'ARCComputingElement: failed to set up proxy', result['Message'] )
       return result
-    endpoints = [arc.Endpoint( "ldap://" + self.ceHost + "/MDS-Vo-name=local,o=grid",
-                               arc.Endpoint.COMPUTINGINFO, 'org.nordugrid.ldapng')]
-    retriever = arc.ComputingServiceRetriever(self.usercfg, endpoints)
-    retriever.wait() # Takes a bit of time to get and parse the ldap information
-    targets = retriever.GetExecutionTargets()
-    ceStats = targets[0].ComputingShare
-    gLogger.debug("Running jobs for CE %s : %s" % (self.ceHost, ceStats.RunningJobs))
-    gLogger.debug("Waiting jobs for CE %s : %s" % (self.ceHost, ceStats.WaitingJobs))
+
+    # Try to find out which VO we are running for.
+    vo = ''
+    res = getVOfromProxyGroup()
+    if res['OK']:
+      vo = res['Value']
 
     result = S_OK()
-    result['RunningJobs'] = ceStats.RunningJobs
-    result['WaitingJobs'] = ceStats.WaitingJobs
     result['SubmittedJobs'] = 0
+    if (vo == '') :
+      # Presumably the really proper way forward once the infosys-discuss WG comes up with a solution
+      # and it is implemented. Needed for DIRAC instances which use robot certificates for pilots.
+      endpoints = [arc.Endpoint( "ldap://" + self.ceHost + "/MDS-Vo-name=local,o=grid",
+                               arc.Endpoint.COMPUTINGINFO, 'org.nordugrid.ldapng')]
+      retriever = arc.ComputingServiceRetriever(self.usercfg, endpoints)
+      retriever.wait() # Takes a bit of time to get and parse the ldap information
+      targets = retriever.GetExecutionTargets()
+      ceStats = targets[0].ComputingShare
+      gLogger.debug("Running jobs for CE %s : %s" % (self.ceHost, ceStats.RunningJobs))
+      gLogger.debug("Waiting jobs for CE %s : %s" % (self.ceHost, ceStats.WaitingJobs))
+      result['RunningJobs'] = ceStats.RunningJobs
+      result['WaitingJobs'] = ceStats.WaitingJobs
+    else:
+      # The system which works properly at present for ARC CEs that are configured correctly.
+      # But for this we need the VO to be known
+      cmd = 'ldapsearch -x -LLL -H ldap://%s:2135 -b mds-vo-name=resource,o=grid "(GlueVOViewLocalID=%s)"' %(self.ceHost, vo.lower())
+      res = shellCall( 0, cmd )
+      if not res['OK']:
+        gLogger.debug("Could not query CE %s - is it down?" % self.ceHost)
+        return res
+      try:
+        ldapValues = res['Value'][1].split("\n")
+        running = [y for y in ldapValues if 'GlueCEStateRunningJobs' in y]
+        waiting = [y for y in ldapValues if 'GlueCEStateWaitingJobs' in y]
+        result['RunningJobs'] = int(running[0].split(":")[1])
+        result['WaitingJobs'] = int(waiting[0].split(":")[1])
+      except IndexError:
+        res = S_ERROR('Unknown ldap failure for site %s' % self.ceHost)
+        res['RunningJobs'] = 0
```

Comment:
This does not really follow the standard that we set ourselves relatively recently (started with DError). S_ERROR should contain only 'OK' and 'Message' as key.